pkg/apis/monitoring/v1/types.go

```diff
@@ -87,24 +87,43 @@ type ProxyConfig struct {
 	// that should be excluded from proxying. IP and domain names can
 	// contain port numbers.
 	//
-	// It requires Prometheus >= v2.43.0.
+	// It requires Prometheus >= v2.43.0 or Alertmanager >= 0.25.0.
 	// +optional
 	NoProxy *string `json:"noProxy,omitempty"`
 	// Whether to use the proxy configuration defined by environment variables (HTTP_PROXY, HTTPS_PROXY, and NO_PROXY).
-	// If unset, Prometheus uses its default value.
 	//
-	// It requires Prometheus >= v2.43.0.
+	// It requires Prometheus >= v2.43.0 or Alertmanager >= 0.25.0.
 	// +optional
 	ProxyFromEnvironment *bool `json:"proxyFromEnvironment,omitempty"`
 	// ProxyConnectHeader optionally specifies headers to send to
 	// proxies during CONNECT requests.
 	//
-	// It requires Prometheus >= v2.43.0.
+	// It requires Prometheus >= v2.43.0 or Alertmanager >= 0.25.0.
 	// +optional
 	// +mapType:=atomic
 	ProxyConnectHeader map[string][]v1.SecretKeySelector `json:"proxyConnectHeader,omitempty"`
 }
 
+// Validate semantically validates the given ProxyConfig.
+func (c *ProxyConfig) Validate() error {
```

Comment:
It diverges from the current validateProxyConfig() function in `pkg/prometheus/promcfg.go` since we don't fail on headers with empty slice. I would recommend a first PR to extract this functionality so we can reuse it in this PR.

Reply:
Thanks. I Got it.
I'll submit the next PR to reuse in `prometheus/promcfg.go` and `prometheus/promcfg.go`